src/renderer/services/eth2/client/interface.ts

```diff
@@ -1,34 +1,32 @@
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
-import {SignedBeaconHeaderResponse, ValidatorIndex, ValidatorResponse, BLSPubkey} from "@chainsafe/lodestar-types";
 import {ILogger} from "@chainsafe/lodestar-utils";
 import {IApiClient} from "@chainsafe/lodestar-validator/lib";
 import {IBeaconApi, IBeaconStateApi} from "@chainsafe/lodestar-validator/lib/api/interface/beacon";
 import {INodeApi} from "@chainsafe/lodestar-validator/lib/api/interface/node";
 import {IValidatorApi} from "@chainsafe/lodestar-validator/lib/api/interface/validators";
 import {IEth2ChainHead} from "../../../models/types/head";
+import {BLSPubkey, SignedBeaconHeaderResponse, ValidatorIndex, ValidatorResponse} from "@chainsafe/lodestar-types";
 
-export interface IEth2BeaconApi extends IBeaconApi {
+export interface ICGEth2BeaconApi extends IBeaconApi {
     state: IBeaconStateApi & {
         getBlockHeader(stateId: "head", blockId: "head" | number | string): Promise<SignedBeaconHeaderResponse>;
         getValidator(stateId: "head", validatorId: string | BLSPubkey | ValidatorIndex): Promise<ValidatorResponse>;
         getValidators(stateId?: "head", validatorIds?: (string | ValidatorIndex)[]): Promise<ValidatorResponse[]>;
     };
 }
-export type IEth2NodeApi = INodeApi;
-export type IEth2ValidatorApi = IValidatorApi;
+export type ICGEth2NodeApi = INodeApi;
+export type ICGEth2ValidatorApi = IValidatorApi;
 
 /**
  * Extends minimal interface(IApiClient) required by lodestar validator
  */
-export interface IGenericEth2Client extends IApiClient {
+export interface ICgEth2ApiClient extends IApiClient {
     config: IBeaconConfig;
-
-    beacon: IEth2BeaconApi;
-
-    validator: IEth2ValidatorApi;
+    beacon: ICGEth2BeaconApi;
+    validator: ICGEth2ValidatorApi;
+    node: ICGEth2NodeApi;
```

Comment:
Isn't there suppose to be events property as well?

Reply:
there is no point in rewriting stuff that already in there as `ICgEth2ApiClient` extends `IApiClient` from `@chainsafe/lodestar-validator` witch internally  extends `ApiClientEventEmitter` witch is `StrictEventEmitter<EventEmitter, IApiClientEvents>`
```typescript
export interface IApiClientEvents {
    beaconChainStarted: () => void;
    [BeaconEventType.BLOCK]: (evt: BeaconBlockEvent["message"]) => void;
    [BeaconEventType.CHAIN_REORG]: (evt: BeaconChainReorgEvent["message"]) => void;
    [BeaconEventType.HEAD]: (evt: HeadEvent["message"]) => void;
    [ClockEventType.CLOCK_SLOT]: (evt: ClockSlotEvent["message"]) => void;
    [ClockEventType.CLOCK_EPOCH]: (evt: ClockEpochEvent["message"]) => void;
}
```
In the end, you cannot overwrite types with `extends` just expand it
```typescript
export interface IApiClient extends ApiClientEventEmitter {
    beacon: IBeaconApi;
    node: INodeApi;
    events: IEventsApi;
    validator: IValidatorApi;
    clock: IBeaconClock;
    genesisValidatorsRoot: Root;
    url: string;
    /**
     * Initiates connection to rpc server.
     */
    connect(): Promise<void>;
    /**
     * Destroys connection to rpc server.
     */
    disconnect(): Promise<void>;
}
```
```typescript
export interface ICgEth2ApiClient extends IApiClient {
    config: IBeaconConfig;
    beacon: ICGEth2BeaconApi;
    validator: ICGEth2ValidatorApi;
    node: ICGEth2NodeApi;

    getVersion(): Promise<string>;
    onNewChainHead(callback: (head: IEth2ChainHead) => void): NodeJS.Timeout;
}
```
will produce
```typescript
interface name {
    config: IBeaconConfig;
    beacon: IBeaconApi & ICGEth2BeaconApi;
    node: INodeApi & ICGEth2NodeApi;
    events: IEventsApi;
    validator: IValidatorApi & ICGEth2ValidatorApi;
    clock: IBeaconClock;
    genesisValidatorsRoot: Root;
    url: string;

    connect(): Promise<void>;
    disconnect(): Promise<void>;
    getVersion(): Promise<string>;
    onNewChainHead(callback: (head: IEth2ChainHead) => void): NodeJS.Timeout;
}
```